xwiki-rendering-api/src/main/java/org/xwiki/rendering/syntax/Syntax.java

```diff
@@ -297,15 +297,12 @@ public int hashCode()
     @Override
     public boolean equals(Object object)
     {
-        if (object == null) {
+        if (object == null && object.getClass() != getClass()) {
```

Comment:
Since Syntax is not really designed to be extended you would replace this whole if by `if (!(object instanceof Syntax))`.

Also I would move it after the `if (object == this)`.

Reply:
okay

Reply:
Done.